### ISSUE TITLE
Unused parameter warnings

### ISSUE THREAD

**Issue by tmayoff** [ETLCPP/etl#930]
We're using this in our codebase and would like to be able to compile with `-Werror` and `-Wunused-parameter`. But are being prevented by this library.

```
../subprojects/etl-20.38.10/include/etl/vector.h:270:25: warning: unused parameter 'n' [-Wunused-parameter]
    void reserve(size_t n)
```

Is there a combination of macros that can silence this warning. Or can this be fixed in this repo?

**Comment by jwellbelove** [comment]
I will be pushing a new release very soon. I'll add the change to it.

**Comment by jwellbelove** [comment]
Fixed 20.39.3
